src/components/GifDetailView/GifDetailView.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Component to show target gif's original file
+ */
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+import favoritesActions from '../../actions/favorites';
+
+import { FavButton } from '../ListGifs/ListGifs.styled';
+import { GifDetailViewStyled } from './GifDetailView.styled';
+
+class GifDetailView extends Component {
+    render() {
+        const { list } = this.props.giphy;
+        const favList = this.props.favorites.originalList.data;
+
+        if (!list || !favList || list.data.length > 1) {
+            return null;
+        }
+        const tgtGif = list.data[0];
+        let isFav = false
+        if (favList) {
+            for(let i= 0, len = favList.length; i<len; i++) {
```

Comment:
a good approach could be to store the list as an object in the reducer, so you just need to ask for the key instead of looping the whole list everytime